src/main/java/ca/uwaterloo/drinkmasterapi/feature/mqtt/controller/MqttClientController.java

```diff
@@ -0,0 +1,53 @@
+package ca.uwaterloo.drinkmasterapi.feature.mqtt.controller;
+
+import ca.uwaterloo.drinkmasterapi.feature.mqtt.service.MqttClientService;
+import com.amazonaws.services.iot.client.AWSIotException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.UnsupportedEncodingException;
+import java.security.NoSuchAlgorithmException;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/mqtt")
+public class MqttClientController {
+
+    @Autowired
+    private MqttClientService mqttClientServiceImpl;
+
+    @GetMapping("/gdpr")
+    public ResponseEntity publishGdprMessage(@RequestParam("locationId") final String locationId) throws AWSIotException {
```

Comment:
do not throw exception here, use try {} catch {} to handle exceptions in the controller, as for the other functions below

Reply:
i will note this down, will change it when I finish debugging from iot core